tests/test_pipelines.py

```diff
@@ -1058,6 +1059,37 @@ def test_from_flax_from_pt(self):
 
         assert np.abs(image_0 - image_1).sum() < 1e-5, "Models don't give the same forward pass"
 
+    @require_compel
+    def test_weighted_prompts_compel(self):
+        from compel import Compel
+
+        pipe = StableDiffusionPipeline.from_pretrained("CompVis/stable-diffusion-v1-4")
+        pipe.scheduler = UniPCMultistepScheduler.from_config(pipe.scheduler.config)
+        pipe.enable_model_cpu_offload()
+        pipe.enable_attention_slicing()
+
+        compel = Compel(tokenizer=pipe.tokenizer, text_encoder=pipe.text_encoder)
+
+        prompt = "a red cat playing with a ball{}"
+
+        prompts = [prompt.format(s) for s in ["", "++", "--"]]
+
+        prompt_embeds = compel(prompts)
+
+        generator = [torch.Generator(device="cpu").manual_seed(33) for _ in range(prompt_embeds.shape[0])]
+
+        images = pipe(
+            prompt_embeds=prompt_embeds, generator=generator, num_inference_steps=20, output_type="numpy"
+        ).images
+
+        for i, image in enumerate(images):
+            expected_image = load_numpy(
+                "https://huggingface.co/datasets/hf-internal-testing/diffusers-images/resolve/main"
+                f"/compel/forest_{i}.npy"
+            )
+
+            assert np.abs(image - expected_image).max() < 1e-3
+
```

Comment:
In the spirit of some of our recent discussions around making sure integration tests are adding value, I don't think we really need this test. 

We're effectively testing 
1. The compel library works
2. The stable diffusion v1-4 model gives consistent outputs
3. The stable diffusion pipeline takes prompt embeds

1 should be covered by the compel libraries own tests
2 should be covered by existing integration tests
3 should be covered by the pipeline mixin unit tests

Reply:
Note that if we remove this test then we also don't have to worry about adding compel as a testing dependency

Reply:
Hmm don't agree here really - here we detect if compel doesn't work for our use case anymore. The test is cheap (9 seconds since model is cached)

What I'm testing here especially is that adding `++` makes a ball appear whereas `--` and `` don't.
Also note that compel is not tested super well, so this is an easy safeguard

Reply:
I agree with @patrickvonplaten, this is mostly a monitoring alert that will trigger if the external library stops working. I think it's also a nice and clear test.